backend/service/xkcd.go

```diff
@@ -0,0 +1,88 @@
+package service
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+	"xkcd/comics/model"
+)
+
+const xkcdURL = "https://xkcd.com/%d/info.0.json"
+
+func GetComic(comicNum int) (*model.Comic, error) {
+    resp, err := http.Get(fmt.Sprintf(xkcdURL, comicNum))
+    if err != nil {
+        return nil, err
+    }
+    defer resp.Body.Close()
+
+    var comic model.Comic
+    if err := json.NewDecoder(resp.Body).Decode(&comic); err != nil {
+        return nil, err
+    }
+
+    return &comic, nil
+}
+
+func GetLatestComicNumber() (int, error) {
+    resp, err := http.Get("https://xkcd.com/info.0.json")
```

Comment:
Using http.Get is fine here as it's a very contrived exercise. However, please do note that in production we'll want to create a custom HTTP client and use that instead so we can provide sensible values for timeouts etc.

See [the go documentation](https://pkg.go.dev/net/http#hdr-Clients_and_Transports) for more details.